openbas-api/src/main/java/io/openbas/service/InjectService.java

```diff
@@ -167,6 +208,636 @@ private List<InjectOutput> execInject(TypedQuery<Tuple> query) {
+                    addFieldsFromExpectations(inject, ruleAttribute,
+                            row, mapTeamByName, expectation, importMapper));
+        });
+        // This is by default at false
+        inject.setAllTeams(false);
+        // The user is the one doing the import
+        inject.setUser(userRepository.findById(currentUser().getId()).orElseThrow());
+        // No exercise yet
+        inject.setExercise(null);
+        // No dependencies
+        inject.setDependsOn(null);
+
+        if(expectation.get() != null) {
+            // We set the expectation
+            ArrayNode expectationsNode = mapper.createArrayNode();
+            ObjectNode expectationNode = mapper.createObjectNode();
+            expectationNode.put("expectation_description", expectation.get().getDescription());
+            expectationNode.put("expectation_name", expectation.get().getName());
+            expectationNode.put("expectation_score", expectation.get().getExpectedScore());
+            expectationNode.put("expectation_type", expectation.get().getType().name());
+            expectationNode.put("expectation_expectation_group", false);
+            expectationsNode.add(expectationNode);
+            inject.getContent().set("expectations", expectationsNode);
+
+            /*if(!inject.getTeams().isEmpty()) {
+                List<InjectExpectation> expectations = inject.getTeams().stream().map(team -> {
+                    InjectExpectation expectationOfTeam = new InjectExpectation();
+                    expectationOfTeam.setDescription(expectation.get().getDescription());
+                    expectationOfTeam.setName(expectation.get().getName());
+                    expectationOfTeam.setScore(expectation.get().getScore());
+                    expectationOfTeam.setType(InjectExpectation.EXPECTATION_TYPE.MANUAL);
+                    expectationOfTeam.setTeam(team);
+                    expectationOfTeam.setExpectationGroup(false);
+                    expectationOfTeam.setInject(inject);
+                    return expectationOfTeam;
+                })
+                    .filter(injectExpectation -> injectExpectation.getScore() != null)
+                    .toList();
+
+                inject.setExpectations(expectations);
+            }*/
+        }
+
+        // We set the scenario
+        inject.setScenario(scenario);
+
+        importTestSummary.setInject(inject);
+        importTestSummary.setInjectTime(injectTime);
+        return importTestSummary;
+    }
+
+    private List<ImportMessage> addFieldsFromExpectations(Inject inject, RuleAttribute ruleAttribute,
+                                                          Row row, Map<String, Team> mapTeamByName,
+                                                          AtomicReference<InjectExpectation> expectation,
+                                                          ImportMapper importMapper) {
+        // If it's a reserved field, it's already taken care of
```

Comment:
Can we validate object before creation ? To be sure it will be validate ?
Or we let the DB return an error ?